### PR TITLE
Convective exit

### PR DESCRIPTION
**Pull request by marinlauber**
Initial commit for convective exit implementation. The 1D convection equation is applied to the `1` direction at the outlet during the **first** step of the predictor-corrector. A global mass flux correction ensures mass conservation on the domain. This mass flux correction imposes a Dirichlet condition (as the standard `BC!(a,A)`) when the time-step is omitted (for all other calls of `BCΔt!(a,A)` basically).

What we need to check:
- [x]  check that a non-uniform profile is convected and scaled such that global mass flux match
- [x]  check that without time-step the mass flux check imposes the same Dirichlet BC on the outlet as the inlet
- [x]  check that the velocity profile at the convective exit is not uniform for the flow behind a cylinder
- [x]  check pressure isocontour on exit
- [x]  check if we can get rid of the original `BC!(a,A)` function (might be an issue here [https://github.com/marinlauber/WaterLily/blob/2d4787c19f1675d756acb4763539223a5999ea19/src/Body.jl#L54](url)) which sets the Homogeneous Neumann for the pressure solver

The time integration works, but there is a mismatch between the pressure field obtained with this and the one we get in Lotus.

### CODE REVIEW

**Comment by weymouth** [comment]
We still need to get create specific exit tests (and get rid of the other file).

File: src/util.jl [hunk 1]
```diff
@@ -150,20 +146,29 @@ condition `a[I,i]=A[i]` is applied to the vector component _normal_ to the domai
...
         else    # Tangential directions, Neumann
             @loop a[I,i] = a[I+δ(j,I),i] over I ∈ slice(N,1,j)
             @loop a[I,i] = a[I-δ(j,I),i] over I ∈ slice(N,N[j],j)
         end
     end
 end
 
+function exitBC!(u,u⁰,U,Δt)
+    N,_ = size_u(u)
+    exit = slice(N.-1,N[1],1,2)               # exit slice excluding ghosts
```
Comment (b-fg) [inline src/util.jl:166]: `exit` is a Julia keyword, would be safer not to override it even if harmless here :)

Comment (weymouth) [reply src/util.jl:166]: Good point!

Comment (b-fg) [reply src/util.jl:166]: replace by `outflow` or `outlet` maybe?

**Pull request merged**